hswidget/cofeditor.go

```diff
@@ -0,0 +1,224 @@
+package hswidget
+
+import (
+	"strconv"
+
+	"github.com/ianling/giu"
+
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2enum"
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2cof"
+
+	"github.com/OpenDiablo2/HellSpawner/hscommon"
+	"github.com/OpenDiablo2/HellSpawner/hscommon/hsenum"
+	"github.com/OpenDiablo2/HellSpawner/hscommon/hsutil"
+)
+
+const (
+	upItemButtonPath     = "3rdparty/iconpack-obsidian/Obsidian/actions/16/stock_up.png"
+	downItemButtonPath   = "3rdparty/iconpack-obsidian/Obsidian/actions/16/stock_down.png"
+	leftArrowButtonPath  = "3rdparty/iconpack-obsidian/Obsidian/actions/16/stock_left.png"
+	rightArrowButtonPath = "3rdparty/iconpack-obsidian/Obsidian/actions/16/stock_right.png"
+)
+
+const (
+	saveCancelButtonW, saveCancelButtonH = 80, 30
+	bigListW                             = 200
+	trueFalseListW                       = 60
+)
+
+// COFEditor contains data necessary do edit cof file
+type COFEditor struct {
+	newCofLayer       *d2cof.CofLayer
+	cof               *d2cof.COF
+	id                string
+	upArrowTexture    *giu.Texture
+	downArrowTexture  *giu.Texture
+	leftArrowTexture  *giu.Texture
+	rightArrowTexture *giu.Texture
+}
```

Comment:
what do you think, maybe I should rename COFViewerWidget into COFEditorWIdget and place it in COFEditorState?